src/manifest/adaptation.ts

```diff
@@ -167,6 +169,10 @@ export default class Adaptation {
       this.isSignInterpreted = parsedAdaptation.isSignInterpreted;
     }
 
+    this.trickModeTracks = trickModeTracks !== undefined ?
+      trickModeTracks.map((track) => new Adaptation(track)) :
+      undefined;
```

Comment:
I don't know if you saw that one, so I put more visibility to it by inserting it in the review.

I don't know how Github will show you that message, so to repeat: 


By reading it, I'm hesitating between what you wrote and:
```js
if (trickModeTracks !== undefined && trickModeTracks.length > 0) {
  this.trickModeTracks = trickModeTracks.map((track) => new Adaptation(track));
}
```
Here we would only define the variable if it is defined, which might be better for people debugging in the console.
This is a very tiny nitpick though, as I like also how you wrote it. You're the judge.

Reply:
I think you're right. Better not to set it again to `undefined`.